package.json

```diff
@@ -7,10 +7,13 @@
   "bugs": "https://github.com/facebook/relay/issues",
   "repository": "facebook/relay",
   "scripts": {
-    "build": "gulp",
+    "build": "gulp dist",
+    "build:clean": "gulp cleanbuild",
+    "build:watch": "gulp watch",
     "jest": "NODE_ENV=test jest \"$@\"",
     "lint": "eslint --max-warnings 0 .",
-    "prepublish": "node node_modules/fbjs-scripts/node/check-dev-engines.js package.json && npm run build",
+    "postinstall": "npm run build",
+    "prepublishOnly": "node node_modules/fbjs-scripts/node/check-dev-engines.js package.json && npm run build:clean",
```

Comment:
This only runs when actually invoking `npm publish` and will perform a clean before the build.